### ISSUE TITLE
Dynamic Modal

### ISSUE THREAD

**Issue by pedrostein** [soflyy/oxygen-bugs-and-features#1498]
**Describe the feature you'd like to see included in Oxygen.**

It would be great if in a repeater, easy post, archive list etc a trigger could be created to open a modal with dynamic content related to that trigger.  

**What are the use cases for this feature?**

This would allow for making a modal to see some extended or full content from whats in the repeater. 
For example a list of real estate objects could show more details like agent address, image gallery etc in a modal and if people show more interest they can have an option to go to the full object description. 

**Examples of this feature or functionality.**

Somehow similar to nice OceanWP nice Quickview option for products in shop pages as you can see in the examples here when hovering over the products but have this option also available in repeaters or easy post :

https://onestore.oceanwp.org/shop/
https://bakery.oceanwp.org/shop/

**Comment by ribenawrath** [comment]
I'm totally in for this. Crocobloks does this very well.

There needs to be a way of building the modal, assinig the data and then linking it to the repeater

**Comment by YammYcoding** [comment]
+1

**Comment by orohellec** [comment]
+1

**Comment by alexeinporiss** [comment]
+1

**Comment by andyduo** [comment]
+1 for this. Have a use case for this now too...

**Comment by richardroscoe** [comment]
+1

**Comment by Rauljauregi** [comment]
+1

**Comment by DrZivil** [comment]
+1

**Comment by BarBoost** [comment]
> Being able to use modals in repeaters is really a very basic need. I'm going to have to use custom PHP in a code block today for something I should have been able to create in a few minutes.

Well, it looks like the guys behind Oxygen abandoned their loyal community to focus on their new builder (breakdance). It seems like a jump ship because of the lifetime licenses.

mlewlp, if you find a solution or can point me to one, I would love to hear about it.

**Comment by pedrostein** [comment]
@BarBoost this would be a great feature to have natively in oxygen. 
As an alternative no-code solution I found that this feature is achievable by using oxyextras 
https://oxyextras.com/docs/lightbox/

**Comment by mlewlp** [comment]
@pedrostein I looked at Oxy Extras' lightbox but unfortunately it wouldn't work for my purpose. But it could definitely fill some of this void in a lot of use cases.

@BarBoost My solution was to set up a modal the way I wanted it, copy the source code, then put that into a custom PHP code block and filled in the content with the values retrieved via PHP. It feels dirty and you can't easily modify the modal's appearance in Oxygen after the fact, but it works.

You'll need to set a unique ID for your triggering element, then use that for the data-trigger-selector attribute on the modal backdrop.

Also, you'll need to include an empty modal without a trigger so Oxygen will add the Javascript needed for your custom modals to trigger.

**Comment by BarBoost** [comment]
@pedrostein Yeah, I know about Oxyextras. It just seems crazy to pay $150 to a third party for a pretty basic feature. I mean it's been two years since you started your post... At least that's my opinion.

@mlewlp Thanks a lot, appreciate the explanation. It's definitely not ideal, but I'll give it a try.

**Comment by dcolumbus** [comment]
I'm revisiting this in case it magically appeared as a feature and I've somehow missed it?
